instrumentation/vaadin-14.2/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/vaadin/VaadinTracer.java

```diff
@@ -0,0 +1,163 @@
+        span.updateName(vaadinServiceContext.vaadinServiceSpanName);
+      }
+    }
+  }
+
+  public Context startRequestHandlerSpan(RequestHandler requestHandler, Method method) {
+    Context current = Context.current();
+    // ignore nested request handlers
+    if (current.get(REQUEST_HANDLER_CONTEXT_KEY) != null) {
+      return null;
+    }
+
+    String spanName = spanNameForMethod(requestHandler.getClass(), method);
+    VaadinServiceContext vaadinServiceContext = current.get(SERVICE_CONTEXT_KEY);
+    if (vaadinServiceContext != null && !vaadinServiceContext.isRequestHandled()) {
+      Span span = ServerSpan.fromContextOrNull(current);
+      if (span != null) {
+        // set server span name to request handler name
+        // we don't really know whether this request handler is going to be the one
+        // that process the request, if it isn't then next handler will also update
+        // server span name
+        span.updateName(spanName);
+      }
+    }
+
+    Context context = super.startSpan(spanName);
+    return context.with(REQUEST_HANDLER_CONTEXT_KEY, Boolean.TRUE);
+  }
+
+  public void endRequestHandlerSpan(Context context, Throwable throwable, boolean handled) {
+    endSpan(context, throwable);
+
+    // request handler returns true when it processes the request, if that is the case then
+    // mark request as handled
+    if (handled) {
+      VaadinServiceContext vaadinServiceContext = context.get(SERVICE_CONTEXT_KEY);
+      if (vaadinServiceContext != null) {
+        vaadinServiceContext.setRequestHandled();
+      }
+    }
+  }
+
+  public void updateServerSpanName(UI ui) {
+    if (ui != null) {
+      Location location = ui.getInternals().getActiveViewLocation();
+      updateServerSpanName(location);
+    }
+  }
+
+  public void updateServerSpanName(Location location) {
+    Context context = Context.current();
+    Span span = ServerSpan.fromContextOrNull(context);
+    if (span != null) {
+      String path = location.getPath();
+      if (!path.isEmpty()) {
+        path = "/" + path;
+      }
+      span.updateName(ServletContextPath.prepend(context, path));
+    }
+  }
+
+  public Context startClientCallableSpan(Class<?> componentClass, String methodName) {
+    return super.startSpan(spanNameForMethod(componentClass, methodName));
+  }
+
+  public Context startRpcInvocationHandlerSpan(
+      RpcInvocationHandler rpcInvocationHandler, Method method, JsonObject jsonObject) {
+    String spanName = spanNameForMethod(rpcInvocationHandler.getClass(), method);
+    if ("event".equals(rpcInvocationHandler.getRpcType())) {
+      String eventType = jsonObject.getString("event");
+      if (eventType != null) {
+        // append event type to make span name more descriptive
+        spanName += "/" + eventType;
+      }
+    }
+    return super.startSpan(spanName);
+  }
+
+  @Override
+  protected String getInstrumentationName() {
+    return "io.opentelemetry.javaagent.vaadin";
```

Comment:
we changed convention recently (#2433)
```suggestion
    return "io.opentelemetry.javaagent.vaadin-14.2";
```